syncopy/io/load_tdt.py

```diff
@@ -0,0 +1,860 @@
+# -*- coding: utf-8 -*-
+# @Author: Diljit Singh Kajal
+# @Date:   2022-04-08 15:00:00
+#
+# load_tdt.py Merge separate TDT SEV files into one HDF5 file
+
+import os
+from datetime import datetime
+import re
+import numpy as np
+from getpass import getuser
+from tqdm.auto import tqdm
+import h5py
+
+# Local imports
+from syncopy.shared.parsers import io_parser, scalar_parser
+from syncopy.shared.errors import SPYWarning, SPYValueError
+from syncopy.shared.tools import StructDict
+import syncopy as spy
+
+
+# --- The user exposed function ---
+
+
+def load_tdt(data_path, start_code=None, end_code=None,
+             subtract_median=False):
+
+    """
+    Imports TDT time series data and meta-information
+    into a single :class:`~syncopy.AnalogData` object.
+
+    An ad-hoc trialdefinition will be attached if
+    both `start_code` and `end_code` are given.
+    Otherwise a single all-to-all trialdefintion
+    is used. Custom trialdefinitions can be done
+    afterwards with :func:`~syncopy.definetrial`.
+
+    All meta-information is stored within the `.info`
+    dict of the  :class:`~syncopy.AnalogData` object.
+
+    PDio related keys:
+        ("PDio_onset", "PDio_offset", "PDio_data")
+
+    Trigger related keys:
+        ("Trigger_code", "Trigger_timestamp", "Trigger_sample")
+
+    Parameters
+    ----------
+    data_path : str
+        Path to the directory containing the `.sev` files
+    start_code : int or None, optional
+        Trigger code defining the beginning of a trial
+    end_code : int or None, optional
+        Trigger code defining the end of a trial
+    subtract_median : bool
+        Set  to `True` to subtract the median from all
+        individual time series
+
+    Returns
+    -------
+    adata : :class:`~syncopy.AnalogData`
+        The tdt data in syncopy format
+
+    Examples
+    --------
+    Load all channels from a source directory `/data/session3/`:
+
+    >>> adata = load_tdt('/data/session3')
+
+    Access the trigger codes and samples:
+
+    >>> trg_code = adata.info['Trigger_code']
+    >>> trg_sample = adata.info['Trigger_sample']
+
+    Load the same data and construct a trialdefinition on the fly:
+
+    >>> adata = load_tdt('/data/session3', start_code=23000, end_code=30020)
+
+    Access the 3rd trial:
+
+    >>> trl_dat = adata.trials[2]
+    """
+
+    io_parser(data_path, isfile=False)
+
+    if start_code is not None and end_code is None:
+        lgl = "trigger codes for both trial start and end"
+        raise SPYValueError(lgl, "end_code", end_code)
+    if end_code is not None and start_code is None:
+        lgl = "trigger codes for both trial start and end"
+        raise SPYValueError(lgl, "start_code", start_code)
+    elif end_code is None and start_code is None:
+        pass
```

Comment:
superfluous I guess, unless you want it here do document the interface. then I would not use `if..., if..elif`but `if, elif, else`.